src/Conversation.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * This file is part of the TelegramBot package.
+ *
+ * (c) Avtandil Kikabidze aka LONGMAN <akalongman@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Longman\TelegramBot;
+
+/**
+ * Class Conversation
+ *
+ * Only one conversation can be active at any one time.
+ * A conversation is directly linked to a user, chat and the command that is managing the conversation.
+ */
+class Conversation
+{
+    /**
+     * All information fetched from the database
+     *
+     * @var array
+     */
+    protected $conversation = null;
+
+    /**
+     * Notes stored inside the conversation
+     *
+     * @var array
+     */
+    protected $protected_notes = null;
+
+    /**
+     * Notes to be stored
+     *
+     * @var array
+     */
+    public $notes = null;
+
+    /**
+     * Telegram user id
+     *
+     * @var int
+     */
+    protected $user_id;
+
+    /**
+     * Telegram chat id
+     *
+     * @var int
+     */
+    protected $chat_id;
+
+    /**
+     * Command to be executed if the conversation is active
+     *
+     * @var string
+     */
+    protected $command;
+
+    /**
+     * Command has been provided
+     *
+     * @var string
+     */
+    protected $command_is_provided;
+
+    /**
+     * Conversation contructor to initialize a new conversation
+     *
+     * @param int    $user_id
+     * @param int    $chat_id
+     * @param string $command
+     */
+    public function __construct($user_id, $chat_id, $command = null)
+    {
+        $this->user_id = $user_id;
+        $this->chat_id = $chat_id;
+        $this->command = $command;
+
+        $this->command_is_provided = (is_null($command)) ? false : true;
```

Comment:
Assigning `true` or `false` explicitly doesn't make sense.
Instead, it's easier to just put:
`$this->command_is_provided = isset($command);` or `... = ($command !== null);`